boa3/analyser/typeanalyser.py

```diff
@@ -0,0 +1,102 @@
+import ast
+from ast import NodeVisitor
+from typing import List, Dict
+
+from boa3.analyser.astanalyser import IAstAnalyser
+from boa3.exception.CompilerError import *
+from boa3.model.module import Module
+
+
+class TypeAnalyser(NodeVisitor, IAstAnalyser):
+    """
+    This class is responsible for the type checking of the code
+
+    :ivar type_errors: a list with the found type errors. Empty by default.
+    """
+
+    def __init__(self, ast_tree: ast.AST):
+        super().__init__(ast_tree)
+        self.type_errors: List[Exception] = []
+        self.modules: Dict[str, Module] = {}
+
+        self.visiting_function_return: Optional[str] = None
+
+        self.visit(self._tree)
+
+    def _log_error(self, error: CompilerError):
+        self.errors.append(error)
+        raise error
+
+    def visit_Module(self, module: ast.Module):
```

Comment:
There is a lot of methods on TypeAnalyser, VisitorCodeGenerator and other classes that are not being called, they inherit another classes and I am not sure if they are being called internally or they will be called on the future. Either way, it would be good if you could make it clear, making "TODO" comments or a general comment about the methods flow calls. Maybe unit testing could be a good idea too.

Reply:
these methods are from the class that they inherit
the methods with the name starting with `visit_` are from the `AST` class and are used to 
walk through the Python abstract syntax tree, so they are being called internally.
I will include some description in each method to clarify what they are.

Reply:
Could you add a description on the class comment? Something similar you told me:
> the methods with the name starting with visit_ are from the AST class and are used to
walk through the Python abstract syntax tree, so they are being called internally